ooniprobe/Utility/TestUtility.mm

```diff
@@ -86,24 +86,31 @@ + (void)downloadUrls:(void (^)(NSArray *))completion {
         path = [NSString stringWithFormat:@"%@&category_codes=%@", path, [categories componentsJoinedByString:@","]];
     }
     NSURL *url = [NSURL URLWithString:path];
-    NSURLSessionDataTask *downloadTask = [[NSURLSession sharedSession] dataTaskWithURL:url completionHandler:^(NSData *data, NSURLResponse *response, NSError *error) {
-        if (!error) {
-            NSDictionary *dic = [NSJSONSerialization JSONObjectWithData:data options:0 error:&error];
-            NSArray *urlsArray = [dic objectForKey:@"results"];
-            NSMutableArray *urls = [[NSMutableArray alloc] init];
-            for (NSDictionary* current in urlsArray){
-                //List for database
-                Url *url = [Url checkExistingUrl:[current objectForKey:@"url"] categoryCode:[current objectForKey:@"category_code"] countryCode:[current objectForKey:@"country_code"]];
-                //List for mk
-                [urls addObject:url.url];
-            }
-            completion(urls);
-        }
-        else {
-            // Fail
-            completion(nil);
-            NSLog(@"error : %@", error.description);
-        }
+    NSURLSessionDataTask *downloadTask = [[NSURLSession sharedSession]
```

Comment:
Since you modified this code, can you also write tests for that (unless they're already present)?

Reply:
this class also needs to be modified by adding `NSURLComponents` either we do everything in this PR or in another PR

Reply:
Okay, since we started changing this code in this PR, let's also add `NSURLComponents` here.